src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,62 @@
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
     @Autowired
     public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers("/register").permitAll()
+                .antMatchers(HttpMethod.GET, "/cinema-halls")
+                .hasAnyRole(USER.name(), ADMIN.name())
+                .antMatchers(HttpMethod.POST, "/cinema-halls")
+                .hasRole(ADMIN.name())
+                .antMatchers(HttpMethod.GET, "/movies")
+                .hasAnyRole(USER.name(), ADMIN.name())
+                .antMatchers(HttpMethod.POST, "/movies")
+                .hasRole(ADMIN.name())
+                .antMatchers(HttpMethod.GET, "/movie-sessions/available")
+                .hasAnyRole(USER.name(), ADMIN.name())
+                .antMatchers(HttpMethod.POST, "/movie-sessions")
+                .hasRole(ADMIN.name())
+                .antMatchers(HttpMethod.PUT, "/movie-sessions/{id}")
+                .hasRole(ADMIN.name())
+                .antMatchers(HttpMethod.DELETE, "/movie-sessions/{id}")
+                .hasRole(ADMIN.name())
+                .antMatchers(HttpMethod.GET, "/orders")
+                .hasRole(USER.name())
+                .antMatchers(HttpMethod.POST, "/orders/complete")
+                .hasRole(USER.name())
+                .antMatchers(HttpMethod.PUT, "/shopping-carts/movie-sessions")
+                .hasRole(USER.name())
+                .antMatchers(HttpMethod.GET, "/shopping-carts/by-user")
+                .hasRole(USER.name())
+                .antMatchers(HttpMethod.GET, "/users/by-email")
+                .hasRole(ADMIN.name())
```

Comment:
you can simplify it, antMatchers() receives varargs
![image](https://user-images.githubusercontent.com/118391929/222960223-e72180f5-ed9d-429b-ae68-9d1befa32f2f.png)